### PR TITLE
Data mapping for Oculus

### PR DESCRIPTION
**Pull request by jawaharrajramaiya**

### CODE REVIEW

File: modules/data-mapper/code/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+module.exports.mapAccount = (account) => {
+    let mappedAccount = {};
+    if (account) {
+        mappedAccount.tId = account.tId;
+        mappedAccount.Name = account.name;
+        if (account.ud) {
+            mappedAccount.TclId = account.ud.tclId;
```
Comment (naveenbezawada) [inline modules/data-mapper/code/index.js]: In all places, Please rename this to tclId from TclId

File: modules/data-mapper/code/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    return mappedAccount;
+};
+
+module.exports.mapVehicle = (vehicle) => {
+    let mappedVehicle = {};
+    if (vehicle) {
+        mappedVehicle.tId = vehicle.tId;
+        mappedVehicle.Name = vehicle.name;
+        if (vehicle.ud) {
+            mappedVehicle.TclId = vehicle.ud.tclId;
```
Comment (naveenbezawada) [inline modules/data-mapper/code/index.js]: Please rename this to tclId from TclId

File: lambda/api/get-customer-asset-by-deviceid/index.js [hunk 3]
```diff
@@ -83,8 +84,12 @@ function getCustomerAssetDetailsByDeviceId(deviceId, stackId, bearerToken) {
             return getAccountDetails(accountTId, bearerToken);
         })
         .then(account => {
-            customerAssetResult.account = account;
-            return getAssociationBetweenDeviceAndVehicle(accountTId, sensorTId, bearerToken);
+            if (account && account.lifeState == "ACTV") {
+                customerAssetResult.account = dataMapper.mapAccount(account);
```
Comment (naveenbezawada) [inline lambda/api/get-customer-asset-by-deviceid/index.js]: Please log the response, as mobile will have only partial data, let us log entire response here for troubleshooting

File: lambda/api/get-customer-asset-by-deviceid/index.js [hunk 4]
```diff
@@ -95,8 +100,8 @@ function getCustomerAssetDetailsByDeviceId(deviceId, stackId, bearerToken) {
             }
         })
         .then(vehicle => {
-            if (vehicle) {
-                customerAssetResult.vehicle = vehicle;
+            if (vehicle && vehicle.lifeState == "ACTV") {
```
Comment (naveenbezawada) [inline lambda/api/get-customer-asset-by-deviceid/index.js]: Please log the response, as mobile will have only partial data, let us log entire response here for troubleshooting

File: modules/data-mapper/code/index.js (hunk 1 above)
Comment (jawaharrajramaiya) [reply modules/data-mapper/code/index.js]: Updated

File: modules/data-mapper/code/index.js (hunk 2 above)
Comment (jawaharrajramaiya) [reply modules/data-mapper/code/index.js]: Updated

File: lambda/api/get-customer-asset-by-deviceid/index.js (hunk 3 above)
Comment (jawaharrajramaiya) [reply lambda/api/get-customer-asset-by-deviceid/index.js]: Oculus response logged in respective function. For above one, data logged `getAccountDetails `function

File: lambda/api/get-customer-asset-by-deviceid/index.js (hunk 4 above)
Comment (jawaharrajramaiya) [reply lambda/api/get-customer-asset-by-deviceid/index.js]: Oculus response logged in respective function. For above one, data logged `getAssetDetails `function

**Pull request merged**